pipes/WDL/workflows/tasks/reports.wdl

```diff
@@ -97,6 +97,31 @@ task plot_coverage {
 }
 
 
+task coverage_report {
+  Array[File]+ mapped_bams
+  Array[File]  mapped_bam_idx # optional.. speeds it up if you provide it, otherwise we auto-index
```

Comment:
Is `mapped_bam_idx ` unused?

Reply:
It is used.. the pysam code when opening the BAM file will look for a similarly named BAI file and fail on the `pileup` command if such a file does not exist. My wrapper code will auto-index if the index doesn't exist, but it saves time to provide these inputs if it already does exist. I don't think we can get around the hard coded assumptions in pysam and samtools about naming BAM and BAI filenames consistently.

Reply:
BTW older versions of the WDL assembly pipeline will just emit mapped BAM files for aligning reads to the final assembly. Very recent versions will also emit a similarly named BAI (because why not). So if they do have the indexes available, they'll be named consistently with the BAMs.. if they don't, they can just skip it.